File: demo/prompt-demo.js
// Simulates a script that blocks on interactive prompts.
import readline from 'node:readline';

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
const ask = (q) => new Promise((res) => rl.question(q, res));
const sleep = (ms) => new Promise((res) => setTimeout(res, ms));

console.log('Preparing installation...');
await sleep(1_500);

const a1 = await ask('Do you want to continue? [y/N] ');
console.log(`  -> got: ${JSON.stringify(a1)}`);
if (!/^y/i.test(a1)) { console.log('Aborted.'); process.exit(1); }

console.log('Installing (simulated)...');
await sleep(2_000);

const a2 = await ask('Enable color output? (yes/no): ');
console.log(`  -> got: ${JSON.stringify(a2)}`);

await ask('Press Enter to finish...');
console.log('Done! All prompts were answered.');
rl.close();
process.exit(0);

File: engine-rs/examples/prompt_demo.rs
//! Simulates a script that blocks on interactive prompts.

use std::io::{BufRead, Write};

fn ask(q: &str) -> String {
    print!("{q}");
    std::io::stdout().flush().unwrap();
    let mut line = String::new();
    std::io::stdin().lock().read_line(&mut line).unwrap();
    line.trim_end_matches(['\r', '\n']).to_string()
}

fn main() {
    println!("Preparing installation...");
    std::thread::sleep(std::time::Duration::from_millis(1_500));

    let a1 = ask("Do you want to continue? [y/N] ");
    println!("  -> got: {a1:?}");
    if !a1.to_lowercase().starts_with('y') {
        println!("Aborted.");
        std::process::exit(1);
    }

    println!("Installing (simulated)...");
    std::thread::sleep(std::time::Duration::from_millis(2_000));

    let a2 = ask("Enable color output? (yes/no): ");
    println!("  -> got: {a2:?}");

    ask("Press Enter to finish...");
    println!("Done! All prompts were answered.");
}
